src/som/vmobjects/SAbstractObject.java

```diff
@@ -24,51 +24,45 @@
 
 package som.vmobjects;
 
-
-
-
 import com.oracle.truffle.api.interop.TruffleObject;
 
-
+import com.oracle.truffle.api.profiles.ValueProfile;
 import som.interpreter.Frame;
 import som.interpreter.Interpreter;
 import som.vm.Universe;
 
 
 public abstract class SAbstractObject implements TruffleObject {
 
-
-
-
   public abstract SClass getSOMClass(Universe universe);
 
+  public abstract ValueProfile getValueProfile();
```

Comment:
This seems very costly.
I would assume there are many many more run-time values than lexical locations. At least for peak-performance purposes. So, putting the value here seems like it would cause a very high overhead.